### PR TITLE
MWPW-161865: adds alt text attribute to background images of merch cards

### PR DESCRIPTION
**Pull request by Axelcureno**
Adds alt attribute to background images of slice merch card.

Resolves: [MWPW-161865](https://jira.corp.adobe.com/browse/MWPW-161865)

M@S Studio test URL: https://main--mas--adobecom.hlx.live/studio.html?milolibs=MWPW-161865--milo--adobecom#path=ccd

**Test URLs:**
- Before: https://main--milo--adobecom.aem.live/docs/library/kitchen-sink/merch-card?martech=off
- After: https://mwpw-161865--milo--adobecom.aem.live/docs/library/kitchen-sink/merch-card?martech=off

### CODE REVIEW

File: libs/features/mas/src/hydrate.js [hunk 1]
```diff
@@ -82,32 +82,39 @@ export function processSubtitle(fields, merchCard, subtitleConfig) {
...
-            case 'ccd-suggested':
-                merchCard.setAttribute(
-                    'background-image',
-                    fields.backgroundImage,
-                );
-                break;
-        }
-    }
+  if (fields.backgroundImage) {
+      switch (variant) {
```
Comment (3ch023) [inline libs/features/mas/src/hydrate.js:91]: i sugges to base decision on backgroundImageConfig.
```
if (backgroundImageConfig.tag) {
    // create tag
}
if (backgroundImageConfig.attribute) {
    merchCard.setAttribute(backgroundImageConfig.attribute, fields.backgroundImage);
}
```

then you only need to add this to ccd-suggested.js AEM_FRAGMENT_MAPPING:
```
backgroundImage: { attribute: 'background-image' },
```

this shifts control to the specific variant, leaving hydrate simple and unaware of ccd or adobe home or any other consumer.

File: libs/features/mas/src/hydrate.js [hunk 2]
```diff
@@ -82,32 +82,39 @@ export function processSubtitle(fields, merchCard, subtitleConfig) {
...
+          case 'ccd-slice':
+              if (backgroundImageConfig) {
+                  const imgAttributes = {
+                      loading: 'lazy',
+                      src: fields.backgroundImage,
+                  };
+                  if (fields.backgroundImageAltText) {
```
Comment (3ch023) [inline libs/features/mas/src/hydrate.js:98]: jira mentions that if backgroundImageAltText is missing we still need to set `role="none"` attribute
>Ensure  elements have alternate text or a role of none or presentation

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles

File: libs/features/mas/src/hydrate.js (hunk 1 above)
Comment (Axelcureno) [reply libs/features/mas/src/hydrate.js]: That's actually a great suggestion. I've made the change and the function is now variant-agnostic. I've also updated UTs accordingly. Thx.

File: libs/features/mas/src/hydrate.js (hunk 2 above)
Comment (Axelcureno) [reply libs/features/mas/src/hydrate.js]: RIght, added.

**Comment by Roycethan** [comment]
@Axelcureno  
1) Plz resolve js conflicts.
2) I get this error when saving the alt-text some times- 
https://main--mas--adobecom.hlx.live/studio.html?milolibs=MWPW-161865--milo--adobecom#path=ccd
image

cc @yesil @3ch023 @npeltier @afmicka

**Comment by Axelcureno** [comment]
> @Axelcureno
> 
> 1. Plz resolve js conflicts.
> 2. I get this error when saving the alt-text some times-
>    https://main--mas--adobecom.hlx.live/studio.html?milolibs=MWPW-161865--milo--adobecom#path=ccd
> 
> image
> cc @yesil @3ch023 @npeltier @afmicka

Fixed @Roycethan , please retest.

**Review by 3ch023 — CHANGES_REQUESTED** [review]
PR shows 278 changes, you branch needs to be fixed.
will DM in slack

**Comment by Axelcureno** [comment]
Closing this one, new PR with these changes: https://github.com/adobecom/milo/pull/3360

**Comment by Axelcureno** [comment]
Closing since it's addressed in https://github.com/adobecom/milo/pull/3289

**Pull request closed without merging**